public/plugin.tsx

```diff
@@ -388,16 +389,18 @@ export class ObservabilityPlugin
     const { dataSourceService, dataSourceFactory } = startDeps.data.dataSources;
 
     // register all s3 datasources
-    dataSourceFactory.registerDataSourceType(S3_DATASOURCE_TYPE, S3DataSource);
-    core.http.get(`${DATACONNECTIONS_BASE}`).then((s3DataSources) => {
-      s3DataSources.map((s3ds) => {
-        dataSourceService.registerDataSource(
-          dataSourceFactory.getDataSourceInstance(S3_DATASOURCE_TYPE, {
-            name: s3ds.name,
-            type: s3ds.connector.toLowerCase(),
-            metadata: s3ds,
-          })
-        );
+    core.http.get(SECURITY_PLUGIN_ACCOUNT_API).then(() => {
```

Comment:
Would it make sense to check if the response of the accounts API call is `401` and only then not trigger the datasource registration? Otherwise if the security plugin doesn't exist it might return a `404` and in this scenario we still go ahead with datasource registration.

Reply:
I changed the code to originally to perform this, but I changed it again to rely on whether the security plugin would be installed so that we can avoid any extra error codes.